### ISSUE TITLE
[RyuJIT/arm32][JitStress=1] Assertion failed 'firstILBlock->hasProfileWeight()'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#14550]
From corefx x86_arm_altjit run, e.g. "@dotnet-bot test Windows_NT x86_arm_altjit Checked corefx_jitstress1".

Happens with `COMPlus_JitStress=1`.

Found in System.Threading.Tasks.Dataflow.Tests:
```
E:\gh\corefx\bin\tests\System.Threading.Tasks.Dataflow.Tests\netcoreapp-Windows_NT-Release-x86>RunTests.cmd E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\
Using E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\ as the test runtime folder.
Executing in E:\gh\corefx\bin\tests\System.Threading.Tasks.Dataflow.Tests\netcoreapp-Windows_NT-Release-x86\
Running tests... Start time: 13:56:33.79

E:\gh\corefx\bin\tests\System.Threading.Tasks.Dataflow.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJit=*
E:\gh\corefx\bin\tests\System.Threading.Tasks.Dataflow.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitNgen=*
E:\gh\corefx\bin\tests\System.Threading.Tasks.Dataflow.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitName=protononjit.dll
E:\gh\corefx\bin\tests\System.Threading.Tasks.Dataflow.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_NoGuiOnAssert=1
E:\gh\corefx\bin\tests\System.Threading.Tasks.Dataflow.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_ContinueOnAssert=0
E:\gh\corefx\bin\tests\System.Threading.Tasks.Dataflow.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitAssertOnNYI=1
E:\gh\corefx\bin\tests\System.Threading.Tasks.Dataflow.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_JitStress=1

E:\gh\corefx\bin\tests\System.Threading.Tasks.Dataflow.Tests\netcoreapp-Windows_NT-Release-x86>call E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\\dotnet.exe xunit.console.netcore.exe System.Threading.Tasks.Dataflow.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonwindowstests  -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Threading.Tasks.Dataflow.Tests
Discovered:  System.Threading.Tasks.Dataflow.Tests
Starting:    System.Threading.Tasks.Dataflow.Tests

Assert failure(PID 16624 [0x000040f0], Thread: 1384 [0x0568]): Assertion failed 'firstILBlock->hasProfileWeight()' in 'System.Collections.Concurrent.ConcurrentQueue`1[KeyValuePair`2][System.Collections.Generic.KeyValuePair`2[System.Int32,System.Int64]]:EnqueueSlow(struct):this' (IL size 122)

    File: e:\gh\coreclr2\src\jit\flowgraph.cpp Line: 13204
    Image: E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\dotnet.exe
```

Full repro:
```
dotnet/coreclr:
E:\gh\coreclr2> build.cmd x86 checked

dotnet/corefx:
E:\gh\corefx> build.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:CoreCLROverridePath=E:\gh\coreclr2\bin\Product\Windows_NT.x86.Checked
E:\gh\corefx> build-tests.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:WithoutCategories=IgnoreForCI /p:PreExecutionTestScript=e:\bugs\SetStressModes.bat

With e:\bugs\SetStressModes.bat:
set COMPlus_AltJit=*
set COMPlus_AltJitNgen=*
set COMPlus_AltJitName=protononjit.dll
set COMPlus_NoGuiOnAssert=1
set COMPlus_ContinueOnAssert=0
set COMPlus_AltJitAssertOnNYI=1
set COMPlus_JitStress=1
```

**Comment by BruceForstall** [comment]
@dotnet/arm32-contrib @briansull

**Comment by BruceForstall** [comment]
@dotnet/arm64-contrib This assert also exists in arm64.

**Comment by briansull** [comment]
Running using    `-parallel none -showprogress`

```
   System.Threading.Tasks.Dataflow.Tests.TransformBlockTests.TestOrdering_Sync_OrderedEnabled(mmpt: -1, dop: 1, EnsureOrdered: null) [STARTING]
   System.Threading.Tasks.Dataflow.Tests.TransformBlockTests.TestOrdering_Sync_OrderedEnabled(mmpt: -1, dop: 1, EnsureOrdered: null) [FINISHED] Time: 0.1323521s
   System.Threading.Tasks.Dataflow.Tests.TransformBlockTests.TestOrdering_Sync_OrderedEnabled(mmpt: -1, dop: 2, EnsureOrdered: null) [STARTING]

Assert failure(PID 29788 [0x0000745c], Thread: 28028 [0x6d7c]): Assertion failed 'firstILBlock->hasProfileWeight()' in 'System.Collections.Concurrent.ConcurrentQueue`1[KeyValuePair`2][System.Collections.Generic.KeyValuePair`2[System.Int32,System.Int64]]:EnqueueSlow(struct):this' (IL size 122)

    File: d:\fxkit\coreclr\src\jit\flowgraph.cpp Line: 13214
    Image: d:\fxkit\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\dotnet.exe
```

**Comment by briansull** [comment]
Fixed with #14801
